qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -314,17 +314,14 @@ def _calculate_clim(self) -> tuple:
 
         log_normalisation = NORM_OPTS[self.norm.currentIndex()] == "Log"
         try:
-            try:
-                masked_data = data[~data.mask]
-                # Use smallest positive value as vmin for log scale, matplotlib takes care of the data skipping part
-                masked_data = masked_data[masked_data > 0] if log_normalisation else masked_data
-                # If any dimension is zero then we have no data in the display area
-                data_is_empty = any(map(lambda dim: dim == 0, masked_data.shape))
-
-                return (0.1, 1.0) if data_is_empty else (masked_data.min(), masked_data.max())
-            except (AttributeError, IndexError):
-                data = data[np.nonzero(data)] if log_normalisation else data
-                return np.nanmin(data), np.nanmax(data)
+            if log_normalisation:
```

Comment:
This check works well, it would be good to put a test in `test_colorbaqr.py` that supplies NaN data to stop a regression.

From the old code it looks like we would expect non-finite data to be masked in `data.mask`?
For image objects `QuadMeshWrapper` and `ModestImage` with the property `get_array_clipped_to_bounds()` (which I thought was used in non-ortho meshes) it looks like the `data.mask` is produced by the call to `ma.masked_invalid` which should do the job, but this isn't called in the fallback `get_array()` (e.g. `SamplingImage`). I don't think we can mask the data returned from `get_array()` as it will mess up the reshape in `get_array_clipped_to_bounds()` - so this is a good solution. If it isn't used elsewhere perhaps we no longer need to mask invalid data in `get_array_clipped_to_bounds()`? But maybe safer to keep as is?